src/SFML/Graphics/Font.cpp

```diff
@@ -639,11 +656,38 @@ Glyph Font::loadGlyph(std::uint32_t codePoint, unsigned int characterSize, bool
         width += 2 * padding;
         height += 2 * padding;
 
-        // Get the glyphs page corresponding to the character size
-        Page& page = loadPage(characterSize);
+        // Get the pages list corresponding to the character size
+        PageList& pages = m_pageLists[characterSize];
 
-        // Find a good position for the new glyph into the texture
-        glyph.textureRect = findGlyphRect(page, {width, height});
+        // Find a page that can fit well the glyph
+        Page* page = NULL;
+        for (PageList::iterator it = pages.begin(); it != pages.end() && !page; ++it)
+        {
+            // Try to find a good position for the new glyph into the texture
+            glyph.textureRect = findGlyphRect(*it, {width, height});
+
+            if (glyph.textureRect.width > 0)
+                page = &*it;
+        }
+
+        // If we didn't find a matching page, create a new one
+        if (!page)
+        {
+            page = &*pages.insert(pages.end(), Page{m_isSmooth});
```

Comment:
Maybe `if (page != nullptr)` for explicitness.

I have the feeling
```cpp
pages.push_back(Page(m_isSmooth));
page = &pages.back();
```
might be clearer than this address-deref-insert-at-end expression.

Last, use constructor syntax `Page(m_isSmooth)`.

Reply:
Why deref then take the address? Can't we just use the return value of `std::unordered_map::insert`?

Reply:
> Why deref then take the address? Can't we just use the return value of `std::list::emplace_back`?
> 
> ```c++
> page = pages.emplace_back(m_isSmooth);
> ```

I can't say why as I didn't originally write this, but using emplace_back seems like a good shout, will do


> Maybe `if (page == nullptr)` for explicitness.

  I don't think comparing to nullptr makes it more explicit, if anything it makes it less intuitive (we're checking if there's not a page, which is exactly how this code reads)


> Last, use constructor syntax `Page(m_isSmooth)`.

why?